build_msvc/msvc-autogen.py

```diff
@@ -42,8 +43,26 @@ def parse_makefile(makefile):
                     lib_sources[current_lib] = []
                     break
 
+def set_custom_paths(qtDir, protocPath):
+    with open('build_msvc\\Directory.build.props', 'r', encoding='utf-8') as rfile:
+        s = rfile.read()
+        if qtDir:
+            s = re.sub(r'(<QtBaseDir>)[^<]+', r'\1%s'%qtDir, s, re.M)
+        if protocPath:
+            s = re.sub(r'(<ProtocPath>)[^<]+', r'\1%s'%protocPath, s, re.M)
```

Comment:
This would change the existing file that tracked by git. I think you could use `@QtBaseDir@` or something else that easy to locate the content and save the template to `Directory.Build.props.in`. Write to `Directory.Build.props` when we run the script. Then, the code could be two simple `replace()` function.

Reply:
Good point, didn't think of that.